### ISSUE TITLE
Use-site diagnostic should be reported for missing type argument in enum containing type

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#45625]
An error should be reported for the reference to `C.F` when compiling c.cs without a.dll.

`csc /t:library a.cs`
```C#
public class A { }
```

`csc /t:library /r:a.dll b.cs`
```C#
public class B<T>
{
    public enum E { F }
}
public class C
{
    public const B<A>.E F = default;
}
```

`csc /r:b.dll c.cs`
```C#
class Program
{
    static void Main()
    {
        const int x = (int)~C.F;
        System.Console.WriteLine(x);
    }
}
```
See [`EnumTests.UseSiteError()`](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Symbol/Symbols/EnumTests.cs#L314)

**Comment by cston** [comment]
cc @AlekseyTs
